packages/component-library/src/components/feedback-indicator/mt-toast/mt-toast-notification.vue

```diff
@@ -0,0 +1,376 @@
+<template>
+  <div
+    class="mt-toast-notification"
+    :class="classes"
+    ref="toastEl"
+    :role="role"
+    :aria-live="ariaLive"
+    tabindex="0"
+    @mouseenter="pauseTimer"
+    @mouseleave="resumeTimer"
+    @keydown.esc="onRemoveToast"
+  >
+    <div class="mt-toast-notification__content">
+      <div class="mt-toast-notification__content-left">
+        <mt-icon
+          v-if="toast.icon"
+          class="mt-toast-notification__icon"
+          :name="toast.icon"
+          size="20px"
+        />
+
+        <div class="mt-toast-notification__message" :class="messageClasses">
+          {{ toast.msg }}
+        </div>
+      </div>
+
+      <div class="mt-toast-notification__content-right">
+        <mt-button v-if="toast.action" @click="onActionClick" size="small">
+          {{ toast.action.label }}
+        </mt-button>
+      </div>
+
+      <div
+        v-if="toast.dismissible || toast.action"
+        class="mt-toast-notification__close-action"
+        aria-hidden="true"
+        data-testid="dismiss-toast"
+        @click="onRemoveToast"
+      >
+        <mt-icon name="solid-times-circle" size="20px" />
+      </div>
+    </div>
+
+    <div v-if="showTimer" class="mt-toast-notification__timer">
+      <div class="mt-toast-notification__timer-loader"></div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import MtIcon from "@/components/icons-media/mt-icon/mt-icon.vue";
+import MtButton from "@/components/form/mt-button/mt-button.vue";
+import {
+  defineProps,
+  toRefs,
+  type PropType,
+  computed,
+  watch,
+  ref,
+  onBeforeUnmount,
+  onMounted,
+} from "vue";
+import type { Toast } from "./mt-toast.vue";
+
+const emit = defineEmits(["remove-toast"]);
+
+const props = defineProps({
+  toast: {
+    type: Object as PropType<Toast>,
+    required: true,
+  },
+  quickDisplay: {
+    type: Boolean,
+    required: false,
+    default: false,
+  },
+  index: {
+    type: Number,
+    required: false,
+    default: NaN,
+  },
+});
+
+const classes = computed(() => {
+  return {
+    "mt-toast-notification__positive": toast.value.type === "positive",
+    "mt-toast-notification__critical": toast.value.type === "critical",
+    "mt-toast-notification__informal": toast.value.type === "informal",
```

Comment:
According to the BEM methtod variants are used with `--`. When we want to target a child we use the underscores.